PMM-6015 building PMM with a python script (#1907)
Percona-Lab/pmm-submodules: README.md

# PMM Submodules

> You can create feature build for PMM in this repo.

Reviewer: ```suggestion
This repository serves the purpose of creating and updating the feature build for PMM Server and PMM Client.
```

> ## How to create feature build

Reviewer: ```suggestion
## How to create a feature build
```

> if you use the same branch name in all repos then you can run:

Reviewer: ```suggestion
If you use the same branch name in all repos then you can run:
```

```console
make create <you branch name>
```
Branches with "you branch name" will be used for all repos or default branch if the branch with this name isn't found in repo.

> ## Q&A

Reviewer: ```suggestion
## FAQ
```

### What if I need custom branch names for some repos?

You can edit ci.yml by hand. For example:
```yaml
deps:
  - name: global
    branch: PMM-0000-fix-everything
  - name: pmm-agent
    branch: PMM-0000-fix-everything-and-even-more
```
Also, you need to add changes to git and push it:
```
git add ci.yml
git commit -m 'use custom branches'
git push
```

> ### What's global repo in ci.yml?

Reviewer 2: having `global` on get deps staging adds more risks then benefits, I'm pretty sure that everybody will use it and it will be harder to figure out from submodules which repos were affected. So it brings the case when different team members might review only PRs requested from them without knowing about other PRs.

Author: We can send comment to PR with the list of branches for example. Or edit the first comment in PR?

Reviewer 2: It's a hack, let's use git to trace changes instead of trying to hide them. If you want to make it easier for developers to rebuild PR in case of new repo added then it's better to create new jenkins job or github action to rewrite ci.yml and commit.

Author: Changed and now global work only on the first run.

> It's name that script will try to find in repo instead default branch.

Reviewer: ```suggestion
It's a branch name that this script will try to find in a repo instead of the default branch (usually called `main` or `PMM-2.0`).
```
